src/main/kotlin/ecommerce/repository/CartStatisticRepository.kt

```diff
@@ -0,0 +1,51 @@
+package ecommerce.repository
+
+import ecommerce.dto.cartStatistics.MembersWhoAddedToCartDTO
+import ecommerce.dto.cartStatistics.TopAddedProductsDTO
+import ecommerce.enums.CartAction
+import ecommerce.model.CartStatistic
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.query.Param
+import org.springframework.stereotype.Repository
+import java.time.LocalDateTime
+
+@Repository
+interface CartStatisticRepository : JpaRepository<CartStatistic, Long> {
+    @Query(
+        """
+    SELECT new ecommerce.dto.cartStatistics.TopAddedProductsDTO(
+        cs.option.name,
+        COUNT(cs),
+        MAX(cs.createdAt)
+    )
+    FROM CartStatistic cs
+    WHERE cs.action = :action
+      AND cs.createdAt >= :since
+    GROUP BY cs.option.name
+    ORDER BY COUNT(cs) DESC, MAX(cs.createdAt) DESC
+    """,
+    )
+    fun findTopProducts(
+        @Param("action") action: CartAction,
+        @Param("since") since: LocalDateTime,
+        pageable: org.springframework.data.domain.Pageable,
```

Comment:
```suggestion
        pageable: Pageable,
```

Why don't we simplify type names such as `org.springframework.data.domain.Pageable` to just `Pageable`? It would make reading the logic a lot easier!

Also, it would be interesting to find a way to reduce unnecessary parts in SQL, such as `new ecommerce.dto.cartStatistics.TopAddedProductsDTO`. There are multiple ways to do it, so feel free to look up some options and tell me what you think about them.

Reply:
Thanks for the tips!
I went ahead and:

- Imported types like Pageable so the method signatures are shorter and easier to read
- Switched JPQL to interface projections instead of those long new `ecommerce.dto.cartStatistics.TopAddedProductsDto(...)` lines

In the service I just map the interfaces to our existing DTOs, so everything outside the repo stays the same.

It definitely feels cleaner now - thanks for pointing it out!